src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotByRequestIdInterface.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotDao;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import java.time.Instant;
+import java.util.UUID;
+
+public interface CreateSnapshotByRequestIdInterface {
+  default StepResult prepareAndCreateSnapshot(
+      FlightContext context,
+      Snapshot snapshot,
+      SnapshotRequestModel snapshotReq,
+      SnapshotBuilderService snapshotBuilderService,
+      SnapshotRequestDao snapshotRequestDao,
+      SnapshotDao snapshotDao,
+      AuthenticatedUserRequest userReq)
+      throws InterruptedException {
+    UUID accessRequestId =
+        snapshotReq.getContents().get(0).getRequestIdSpec().getSnapshotRequestId();
+    SnapshotAccessRequestResponse accessRequest = snapshotRequestDao.getById(accessRequestId);
+
+    UUID dataReleaseSnapshotId = accessRequest.getSourceSnapshotId();
+    Snapshot dataReleaseSnapshot = snapshotDao.retrieveSnapshot(dataReleaseSnapshotId);
+    // get the underlying dataset for the snapshot
+    Dataset dataset =
+        dataReleaseSnapshot.getSnapshotSources().stream()
+            .findFirst()
+            .orElseThrow(
+                () -> new IllegalArgumentException("Snapshot does not have a source dataset"))
+            .getDataset();
+    // gets pre-existing asset on the dataset
+    // TODO: create custom asset DC-1016
+    AssetSpecification assetSpecification =
+        dataset.getAssetSpecificationByName("concept_asset").orElseThrow();
```

Comment:
The string `concept_asset` is in two places, could it have a constant? Or is there another way to avoid the duplication between this code and the code in `SnapshotService`?

Reply:
yes I will make a static string in SnapshotService. This will be removed soon in the follow on PR that implements asset creation.